src/App.tsx

```diff
@@ -1,14 +1,76 @@
-import React from 'react';
+import {
+  useMemo,
+  useState,
+  useCallback,
+  useEffect,
+} from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { DropdownMenu } from './DropdownMenu';
+import { Person } from './types/Person';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+const debounce = <T extends string[]>(
+  f: (...args: T) => void,
+  delay: number,
+) => {
+  let timerId: NodeJS.Timeout;
+
+  return (...args: T) => {
+    clearTimeout(timerId);
+    timerId = setTimeout(f, delay, ...args);
+  };
+};
+
+export const App = () => {
+  const [query, setQuery] = useState<string>('');
+  const [appliedQuery, setAppliedQuery] = useState<string>('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [showSuggestions, setShowSuggestions] = useState(false);
+
+  useEffect(() => {
+    if (!query.trim()) {
+      setShowSuggestions(false);
+    }
+  }, [query]);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, 1000),
+    [],
+  );
+
+  const handleQueryOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setQuery(value);
+    applyQuery(value);
+    setShowSuggestions(false);
+  };
+
+  const onSelected = (person: Person) => {
+    setQuery(person.name);
+    setSelectedPerson(person);
+    setShowSuggestions(false);
+  };
+
+  const visiblePeople = useMemo(() => {
+    setShowSuggestions(true);
+
+    const normalizePersonName = (human: Person) => {
+      return human.name.toLowerCase().trim();
+    };
+
+    return peopleFromServer.filter(
+      person => normalizePersonName(person)
+        .includes(appliedQuery.toLowerCase()),
+    );
```

Comment:
let's remove normalizePersonName function.

```suggestion
    return peopleFromServer.filter(
      person => {
        const normalizedPersonName = person.name.toLowerCase().trim()
      
        return normalizedPersonName.includes(appliedQuery.toLowerCase())
     });
```